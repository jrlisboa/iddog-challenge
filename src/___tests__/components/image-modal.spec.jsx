/**
 * @jest-environment jsdom
 */

import React from 'react';
import { mount } from 'enzyme';
import ImageModal from '../../components/image-modal';

describe('ImageModal component', () => {
    const list = [
        'https://images.dog.ceo/breeds/pug/n02110958_10.jpg',
        'https://images.dog.ceo/breeds/pug/n02110958_10186.jpg',
        'https://images.dog.ceo/breeds/pug/n02110958_10193.jpg',
    ];
    const onChangeMock = jest.fn();
    const component = mount(
        <ImageModal
            list={list}
            selected={0}
            change={onChangeMock}
            category="pug" />,
    );

    it('Should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('Should render correct image', () => {
        expect(component.find('img').props().src).toBe(list[0]);
    });

    it('Should call the change function with undefined', () => {
        component.simulate('click');
        expect(onChangeMock.mock.calls.length).toBe(1);
        expect(onChangeMock).toHaveReturnedWith(undefined);
    });
});
