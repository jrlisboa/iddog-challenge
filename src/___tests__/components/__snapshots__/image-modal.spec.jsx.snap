// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageModal component Should match snapshot 1`] = `ReactWrapper {}`;
